### PR TITLE
Update Map zoom_control variable to also allow string to set position

### PR DESCRIPTION
**Pull request by berrfred**
This is referred to issue [#1865](https://github.com/python-visualization/folium/issues/1865)

### CODE REVIEW

**Comment by berrfred** [comment]
This is referred to issue [#1865](https://github.com/python-visualization/folium/issues/1865)

File: folium/folium.py [hunk 1]
```diff
@@ -150,8 +150,8 @@ class Map(JSCSSMixin, MacroElement):
         Forces Leaflet to not use hardware-accelerated CSS 3D
         transforms for positioning (which may cause glitches in some
         rare environments) even if they're supported.
-    zoom_control : bool, default True
-        Display zoom controls on the map.
+    zoom_control : bool or position string, default True
+        Display zoom controls on the map, eventually specifying its position.
```
Comment (Conengmo) [inline folium/folium.py:154]:
```suggestion
        Display zoom controls on the map. The default `True` places it in the top left corner. 
        Other options are 'topleft', 'topright', 'bottomleft' or 'bottomright'.
```

File: folium/folium.py [hunk 2]
```diff
@@ -284,10 +288,17 @@ def __init__(
         self.crs = crs
         self.control_scale = control_scale
 
+        # Zoom control position specified ?
+        if isinstance(zoom_control, str):
+            self.zoom_control_position = True
+            self.zoom_control = zoom_control
```
Comment (Conengmo) [inline folium/folium.py:294]: maybe useful to add a check here for valid values? Now if a users makes an error, they just get a grey screen. For example something like:

```suggestion
            self.zoom_control = zoom_control
            if zoom_control not in {'topleft', 'topright', 'bottomleft', 'bottomright'}:
                raise ValueError("Incorrect value for `zoom_control`, choose from 'topleft', 'topright', 'bottomleft' or 'bottomright'.")
```

**Review by Conengmo — CHANGES_REQUESTED** [review]
Implementation is good! I added two comments about the four possible string values. I'd like to make sure users won't make mistakes in these. I hope you can address these comments, afterwards this is good to go!

**Comment by Conengmo** [comment]
Somethings going on with the tests, it's probably not related to this PR.

File: folium/folium.py (hunk 2 above)
Comment (berrfred) [reply folium/folium.py:300]: I had considered introducing a check on valid position values but looking at various plugins I could not find such a check so I thought it was considered as unnecessary ...

Comment (Conengmo) [reply folium/folium.py:305]: I understand, this code base grew quite organically over time, so it's not always consistent. Thanks for putting it in!

**Review by Conengmo — APPROVED** [review]

**Comment by Conengmo** [comment]
Thanks for your contribution @berrfred!

**Pull request merged**